tests/unit/test_main.py

```diff
@@ -0,0 +1,70 @@
+import logging
+import sys
+
+from unittest.mock import patch, Mock
+
+from auditwheel.main import main
+
+
+def test_non_linux_platform(monkeypatch, capsys):
+    # GIVEN
+    monkeypatch.setattr(sys, "platform", "darwin")
+
+    # WHEN
+    retval = main()
+
+    # THEN
+    assert retval == 1
+    captured = capsys.readouterr()
+    assert captured.out == "Error: This tool only supports Linux\n"
```

Comment:
What's the goal of a test like this? You're testing that the code is exactly what's written. If we somehow add support for running on Darwin, or if we switch to a different way of checking the platform instead of `sys.platform`, we just have some extra work to change or discard the test as well.

Rather than trying to execute every line of code in the tests, I'd suggest we focus on the predictive value of the tests. If a test fails, what's the chance that it's identifying a real bug in the code under test? What's the chance we need to change the test to match changes in the code? If the second is much more likely than the first (like in this test, as far as I can see), it's a waste of time.

If it's important to get "100% coverage", I'd prefer to [exclude sections from coverage](https://coverage.readthedocs.io/en/v4.5.x/excluding.html) instead of writing hyper-specific tests just to make those lines run.

Reply:
As we've agreed previously on not making it possible to run `auditwheel` on non-Linux platforms, I think it makes sense to have a test for it.

Would you be OK with having only one assert for the return value and no validation on the error message? At that point all we do is to set the platform to `linux` and make sure the tool errors out. I think this is better than simply excluding the logic from coverage and I don't see any other way to validate this.

Reply:
I think it's better without checking the exact error message. It still seems a fairly pointless test, though: all the situations where I can imagine it failing would involve us fixing the test. It's really hard to imagine introducing a bug in the code which this would pick up.

I guess I don't think 100% test coverage is a useful goal. I like coverage information as a way of spotting chunks of code which aren't tested, but I think it's then a judgment call if you want to test them, and for things like this it's fine not to.

Reply:
I would drop the string check as well.

Reply:
> I would drop the string check as well.

That wasn't clear at all.
While checking the content of the message is unnecessary, I think we should at least check there's something (we don't check it's meaningful but at least something was written out).
```
captured = capsys.readouterr()
assert len(captured.out) > 0
```